src/main/webapp/linkWizard/SelectSampleConsent.js

```diff
@@ -187,6 +192,27 @@ export const SelectSampleConsent = hh(class SelectSampleConsent extends Componen
     return sampleCollectionList;
   };
 
+  handleCheck = () => {
+    this.setState(prev => {
+      prev.onGoingProcess = !this.state.onGoingProcess;
+      prev.endDate = null;
+      return prev;
+    }, () => this.props.updateDateRange(this.state.onGoingProcess, this.state.endDate, this.state.startDate));
+  };
+
+  handleChange = (id) => (date) => {
+    this.setState(prev => {
+      prev[id] = date;
+      return prev;
+    }, () => {
+      if (id == 'startDate' || id == 'endDate') {
+        this.props.updateDateRange(this.state.onGoingProcess, this.state.endDate, this.state.startDate);
+      } else {
+        this.props.updateForm(this.state.sampleCollection, id);
+      }      
```

Comment:
It looks like the only way this is called is via the two `InputFieldDatePicker` references ... is this else condition reachable?

Reply:
One way to avoid having to do these kinds of else checks is to have differently named handlers for the different things that do the handling, that way the same function doesn't have to do double-duty.

Reply:
I agree, I'll create another handler